lib/mixlib/install/options.rb

```diff
@@ -25,15 +25,20 @@ class Install
     class Options
       class InvalidOptions < ArgumentError; end
 
-      attr_reader :options, :errors, :original_platform_version
+      attr_reader :options, :errors, :original_platform_version, :original_architecture
 
       SUPPORTED_ARCHITECTURES = %w{
+        amd64
         i386
+        i686
+        i86pc
         powerpc
         ppc64
         ppc64le
         s390x
         sparc
+        sun4u
+        sun4v
         x86_64
       }
```

Comment:
So I see the issue here now. We don't really want to list these architectures we internally map (`amd64`, `i686`, `i86pc`, `sun4u`, and `sun4v`) as "supported" BUT we want to provide a nice UX and do the right thing. I wonder if the proper place for this automatic mapping to occur is actual in `chef_ingredient` (you could use `Util.normalize_architecture`) as the place we'll need to do this mapping is a Chef recipe as the architecture will come from Ohai. So essentially mixlib-install holds the mapping logic BUT we execute the mapping in `chef_ingredient` before making a call over to mixlib-install.